pkg/controller/controller.go

```diff
@@ -66,10 +66,10 @@ func equal(c1, c2 model.Config) bool {
 // 1. Converts the current and desired slices into a map for quick lookup
 // 2. Loops through the desired slice of items and identifies items that need to be created/updated
 // 3. Loops through the current slice of items and identifies items that need to be deleted
-func (c *Controller) computeChangeList(current []model.Config, desired []model.Config) []*processor.Item {
+func computeChangeList(current []model.Config, desired []model.Config, errHandler processor.OnErrorFunc) []*processor.Item {
```

Comment:
I am thinking we make the on error function part of controller and call c.errHandler directly instead of passing it.